From ab4a9b4bdfbb665c8ca72aab63a03fd716603069 Mon Sep 17 00:00:00 2001
From: Ming Jiang <quic_mjiang@quicinc.com>
Date: Mon, 24 Feb 2025 02:02:14 -0800
Subject: [PATCH] imx8mq.dtsi Add wlan mhi resource

Add mhi resource for cnss hsp detection
Add wlan_en_gpio resource for imx8mqevk 3G ddr board

Change-Id: I19d3f582ba007b80f08f8c4ef332ed8b8ab8155f
Upstream-Status: Pending
Signed-off-by: Ming Jiang <quic_mjiang@quicinc.com>
---
 arch/arm64/boot/dts/freescale/imx8mq.dtsi | 40 +++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mq.dtsi b/arch/arm64/boot/dts/freescale/imx8mq.dtsi
index f67d31d6de65..2bfe1066d615 100644
--- a/arch/arm64/boot/dts/freescale/imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mq.dtsi
@@ -260,6 +260,46 @@ pmu {
 		interrupt-parent = <&gic>;
 	};
 
+	mhi: qcom,mhi {
+		compatible = "qcom,mhi";
+		status = "okay";
+	};
+
+	ipc_router: qcom,ipc_router_external_modem_xprt2 {
+		compatible = "qcom,ipc_router_mhi_xprt";
+		qcom,ch-name = "IPCRTR";
+		qcom,out-chan-id = <20>;
+		qcom,in-chan-id = <21>;
+		qcom,xprt-remote = "external-wlan";
+		qcom,xprt-linkid = <1>;
+		qcom,xprt-version = <3>;
+		status = "okay";
+	};
+
+	diag_mhi: qcom,diag-mhi {
+		compatible = "qcom,diag-mhi";
+		status = "okay";
+	};
+
+	cnss2: qcom,cnss-qca-converged {
+		compatible = "qcom,cnss-qca-converged";
+		wlan-en-gpio = <&gpio5 10 GPIO_ACTIVE_HIGH>;
+		qcom,converged-dt;
+		qcom,bus-type=<0>;
+
+		status = "okay";
+
+		/* Support Hasting Device */
+		chip_cfg@1 {
+			supported-ids = <0x1101>;
+		};
+
+		/* Support Hasting Prime Device */
+		chip_cfg@3 {
+			supported-ids = <0x1103>;
+		};
+	};
+
 	psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
-- 
2.34.1

